lib/ansible/executor/task_executor.py

```diff
@@ -673,8 +673,14 @@ def _evaluate_failed_when_result(result):
             # if we didn't skip this task, use the helpers to evaluate the changed/
             # failed_when properties
             if 'skipped' not in result:
-                _evaluate_changed_when_result(result)
-                _evaluate_failed_when_result(result)
+                try:
+                    condname = 'changed'
```

Comment:
I wonder if we should put each `_evaluate_*_when_result` call into separate `try..except` block for the (I guess rare) case when both `changed_when` and `failed_when` fail. Then we could include the failure reasons for both in the task result at once.

Reply:
Though that would violate "fail early", so meh.